IntimateMerger Analytics Adapter : initial release (#21)
IntimateMerger/Prebid.js: modules/imAnalyticsAdapter.md

# Overview

```
Module Name: IM Analytics Adapter
Module Type: Analytics Adapter
```

# Description

> Analytics adapter for Intimate Merger platform. This adapter tracks auction events and bid won data for analytics purposes.

Reviewer: ```suggestion
Analytics adapter for Intimate Merger. This adapter tracks auction events and bid won data for analytics purposes.
```

> 

Reviewer: https://github.com/IntimateMerger/Prebid.js/blob/master/modules/33acrossAnalyticsAdapter.md
https://docs.prebid.org/dev-docs/modules/userid-submodules/imuid.html

ここあたりを参考に、
- プラポリのURLを記載（可能なら、同意したとみなされるよって、書けるとよいかな）
- CID はアカウントマネージャーから取得するか、LP から申し込んでね。

If you are an existing publisher and you already use a ... は、IM-UID と同じものが使えるよ。

みたいな感じで書いておくとよいかもと思ったりしました。

The adapter monitors the following Prebid.js events:
- `AUCTION_INIT`: Tracks page views and auction initialization
- `BID_WON`: Tracks winning bids with metadata

> - `AUCTION_END`: Triggers batch sending of won bids data

Reviewer: なくてもいいかもと思いました。（だいたいみんなやってること一緒だから）

# Configuration Options

| Parameter | Type | Required | Default | Description |
|-----------|------|----------|---------|-------------|
| `cid` | number | No | 5126 | Client ID for API endpoint |

> | `bidWonTimeout` | number | No | 800 | Timeout in milliseconds before sending batched won bids |

Reviewer: ```suggestion
|-----------|------|----------|---------|-------------|
| `cid` | number | No | 5126 | Client ID for API endpoint |
| `bidWonTimeout` | number | No | 1500 | Timeout in milliseconds before sending batched won bids |
```

# Overview

```
Module Name: IM Analytics Adapter
Module Type: Analytics Adapter
```

#### About

> This analytics adapter collects data about auction events and bid won data on your site for analytics purposes.

Reviewer: @eknis 指示して書いて頂いておきながら、いざできたの読んでみると、
若干微妙に感じてしまいました・・・ごめんなさい x2

実際は analytics purpose （wonBid などの分析）より 収益化（log を売ること）だけのためなのだが、ここには書けないので・・・・

ということで、Simpleに

Analytics adapter for intimatemerger.com.  

or 

Analytics Adapter for IM-DMP.

くらいでよいかもと思いました。

Author: `Analytics Adapter for IM-DMP.`
でいきます

The publisher must work with our account management team to obtain a
Customer ID (CID) and enable Analytics for their account.
To get a CID, you can reach out to your account manager or sign up via

> <https://lp.intimatemerger.com/im-uid>.

Reviewer: いざ書いて頂くと冗長に見えてしまいました。ごめんなさい。ごめんなさい。
（2個書くと重いっすね・・・・）

Please visit [intimatemerger.com/im-uid](https://intimatemerger.com/im-uid) and request your Customer ID to get started.

くらいにで十分かも。

If you are an existing publisher and you already use
[IM-UID](https://docs.prebid.org/dev-docs/modules/userid-submodules/imuid.html),

> you can use the same CID for this analytics adapter.

Reviewer: ```suggestion
you can use the same Customer ID for this analytics adapter.
```

短縮しなくても良さげな気がしました。

Author: 文言の変更でcid表記が減ったので妥当だと思いました。修正します

By enabling this adapter, you agree to Intimate Merger's privacy policy at

> <https://corp.intimatemerger.com/privacypolicy-en/>.

Reviewer: ```suggestion
<https://corp.intimatemerger.com/privacypolicy/>.
```

まぁ、基本に日本向けなので、IM-UID Module にあわえてこちらにしておきましょうか。

#### Analytics Options

| Parameter | Scope | Type | Default | Description |
|-----------|-------|------|---------|-------------|
| `cid` | optional | number | 5126 | Customer ID provided by Intimate Merger |

> | `bidWonTimeout` | optional | number | 1500 | Timeout in milliseconds before sending batched won bids |

Reviewer: cid は required の方がよいかなと！（IM-UID Module と一緒で、5126 は あくまで fallback ）

- Parameter
- Scope
- Type
- Example
- Description

にするのがよいっすかね？

```suggestion
| Parameter | Scope | Type | Example | Description |
|-----------|-------|------|---------|-------------|
| `cid` | required | number | 5126 | The Customer ID provided by Intimate Merger. |
| `bidWonTimeout` | optional | number | 1500 | Wait time in milliseconds before sending batched requests. (Default: 1500)  |

```

ふと書いてて思ったのですが `WaitTimeout` とかの方がそれっぽかったりします？
（そこまでこだわりはないのですが・・・・）

Author: 他見てみたんですが

アダプター | オプション名 | 
-- | -- | 
imAnalyticsAdapter | bidWonTimeout |
atsAnalyticsAdapter | bidWonTimeout | 
33acrossAnalyticsAdapter | timeout | 

でした
確かに利用者が詳細気にするかというとそこまで説明的でなくてもいいかなと思います

#### Example Configuration

##### Basic Configuration

```javascript
pbjs.enableAnalytics({
    provider: 'imAnalytics'
});
```

##### Configuration with Custom CID

```javascript
pbjs.enableAnalytics({
    provider: 'imAnalytics',
    options: {
        cid: 1234
    }
});
```

> ##### Configuration with Custom Timeout

Reviewer: これだけ項目数が少ないので、1つでコメントつきで良さそうな。

> 
> ```javascript
> pbjs.enableAnalytics({
>     provider: 'imAnalytics',
>     options: {
>         cid: 1234,
>         bidWonTimeout: 1000  // 1 second
>     }
> });

Reviewer: ```suggestion
pbjs.enableAnalytics({
    provider: 'imAnalytics',
    options: {
        /* Required: Customer ID */
        cid: 5126,
        /* Optional: Wait 2 seconds */
        bidWonTimeout: 2000
    }
});
```

シンプルな設定なので、IM-UIDにあわせて、このくらいでよいのかな〜と。
